parbase/FairParRootFileIo.cxx

```diff
@@ -149,49 +149,41 @@ Bool_t FairParRootFileIo::open(const TList* fnamelist, Option_t* option, const T
 {
     TDatime currentDate;
     TString newParFileName = "";
-    TFile* newParFile = 0;
+    std::unique_ptr<TFile> newParFile;
 
     TObjString* string;
     TListIter myIter(fnamelist);
 
-    TKey* inpKey;
-
-    TFile* inFile;
-
-    Int_t nofFiles = 0;
     while ((string = static_cast<TObjString*>(myIter.Next()))) {
-        inFile = TFile::Open(string->GetString().Data());
+        std::unique_ptr<TFile> inFile(TFile::Open(string->GetString().Data()));
         if (!inFile) {
             cout << "-W- File \"" << string->GetString().Data() << "\" does not exist" << endl;
             continue;
         }
 
-        if (nofFiles == 0) {
+        if (!newParFile) {
             newParFileName = string->GetString();
             newParFileName.Replace(newParFileName.Last('/') + 1, newParFileName.Length(), "");
             newParFileName =
                 Form("%sallParams_%d_%06d.root", newParFileName.Data(), currentDate.GetDate(), currentDate.GetTime());
-            newParFile = TFile::Open(newParFileName.Data(), "RECREATE");
+            newParFile.reset(TFile::Open(newParFileName.Data(), "RECREATE"));
         }
 
         TList* inputKeys = static_cast<TList*>(inFile->GetListOfKeys());
 
         TListIter keyIter(inputKeys);
+        TKey* inpKey;
         while ((inpKey = static_cast<TKey*>(keyIter.Next()))) {
             TObject* tempObj = inFile->Get(inpKey->GetName());
 
             newParFile->cd();
             tempObj->Write();
         }
-        inFile->Close();
```

Comment:
Why the files are not closed any longer? Currently this is done in the destructor of TFile but it this something one should rely on?

Reply:
I'd say, that this is the whole purpose of a destructor?

The only real problem I can see: `Close` is a virtual method. Calling a virtual method from a destructor is highly not reommended / possibly doesn't do what you would expect it to do.

But then again: Why do we have a destructor, when it is not cleaning up properly?

@dennisklein any thoughts on this?

Reply:
My first thought (before @fuhlig1's question) was to absolutely rely on TFile object destruction to also close the file. But now I am puzzled

@fuhlig1 Can you concretize your concern?

Reply:
Since the file is properly closed in the destructor I am fine with the commit.